src/test/java/fi/otavanopisto/kuntaapi/test/server/integration/ptv/ServiceLocationServiceChannelInTestsIT.java

```diff
@@ -0,0 +1,149 @@
+package fi.otavanopisto.kuntaapi.test.server.integration.ptv;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import com.jayway.restassured.http.ContentType;
+
+import fi.metatavu.kuntaapi.server.rest.model.ServiceLocationServiceChannel;
+import fi.metatavu.ptv.client.model.V7VmOpenApiServiceLocationChannel;
+import fi.metatavu.ptv.client.model.V7VmOpenApiServiceLocationChannelInBase;
+import fi.otavanopisto.kuntaapi.server.persistence.model.clients.AccessType;
+import fi.otavanopisto.kuntaapi.server.persistence.model.clients.ClientOrganizationPermission;
+import fi.otavanopisto.kuntaapi.test.AbstractPtvMocker;
+
+public class ServiceLocationServiceChannelInTestsIT extends AbstractPtvInTest {
+
+  /**
+   * Starts WireMock
+   */
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(getWireMockPort());
+  
+  @Before
+  public void beforeTest() throws InterruptedException {
+    getPtvOrganizationMocker().mock(TestPtvConsts.ORGANIZATIONS);
+    getPtvServiceMocker().mock(TestPtvConsts.SERVICES);    
+    getPtvServiceChannelMocker().mock(TestPtvConsts.SERVICE_CHANNELS);
+
+    startMocks();
+    
+    waitApiListCount("/organizations", 3);
+    waitApiListCount("/serviceLocationServiceChannels", TestPtvConsts.SERVICE_LOCATION_SERVICE_CHANNELS.length);
+    waitApiListCount("/services", TestPtvConsts.SERVICES.length);
+  }
+  
+  @Test
+  public void updateServiceLocationServiceChannelUnauthorized() throws JsonParseException, JsonMappingException, IOException, InterruptedException {
+    ServiceLocationServiceChannel kuntaApiResource = getServiceLocationServiceChannel(0, TestPtvConsts.SERVICE_LOCATION_SERVICE_CHANNELS.length);
+    
+    givenReadonly()
+      .body(kuntaApiResource)
+      .contentType(ContentType.JSON)
+      .put("/serviceLocationServiceChannels/{kuntaApiChannelId}", kuntaApiResource.getId())
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "/serviceLocationServiceChannels/{kuntaApiChannelId}" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)